From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 6 Feb 2016 13:10:28 +0100
Subject: [PATCH] {BUG} drivers/power/lp8788-charger: fix uninitialized return
 value

clang:
drivers/power/lp8788-charger.c:461:14: error: variable 'found' is
      used uninitialized whenever 'for' loop exits because its condition is false
      [-Werror,-Wsometimes-uninitialized]
        for (i = 0; i < pchg->num_irqs; i++) {
                    ^~~~~~~~~~~~~~~~~~
drivers/power/lp8788-charger.c:469:9: note: uninitialized use occurs
      here
        return found;
               ^~~~~
drivers/power/lp8788-charger.c:461:14: note: remove the condition if
      it is always true
        for (i = 0; i < pchg->num_irqs; i++) {
                    ^~~~~~~~~~~~~~~~~~
drivers/power/lp8788-charger.c:458:12: note: initialize the variable
      'found' to silence this warning
        bool found;
                  ^
                   = 0
---
 drivers/power/lp8788-charger.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/power/lp8788-charger.c b/drivers/power/lp8788-charger.c
index f5a48fd68b01..7321b727d484 100644
--- a/drivers/power/lp8788-charger.c
+++ b/drivers/power/lp8788-charger.c
@@ -455,7 +455,7 @@ static void lp8788_charger_event(struct work_struct *work)
 
 static bool lp8788_find_irq_id(struct lp8788_charger *pchg, int virq, int *id)
 {
-	bool found;
+	bool found = false;
 	int i;
 
 	for (i = 0; i < pchg->num_irqs; i++) {
-- 
